app/controllers/Dasher.scala

```diff
@@ -34,7 +34,11 @@ object Dasher extends LilaController {
         I18nKeys.boardSize,
         I18nKeys.pieceSet
       ), ctx.lang
-  )
+  ) ++ Env.i18n.jsDump.keysToObject(
+      // the language settings should never be in a totally foreign language
+      List(I18nKeys.language),
+      if (I18nLangPicker.allFromRequestHeaders(ctx.req).has(ctx.lang)) ctx.lang else enLang
```

Comment:
here you could go farther. If the browser doesn't accept `ctx.lang`, then use the browser favourite language that lichess knows how to translate. If none is found, then default to english.